File: windows/src/SoliBee.Core/Models/HoneycombStats.cs
using System;
using System.Collections.Generic;

namespace SoliBee.Core.Models;

public class HoneycombStats
{
    public int GamesPlayed { get; set; }
    public int MatchesWon { get; set; }
    public int MatchesLost { get; set; }
    public int MatchesDrawn { get; set; }
    public int CardsCaptured { get; set; }
    public int CurrentWinStreak { get; set; }
    public int LongestWinStreak { get; set; }
    public int FlawlessVictories { get; set; }
    public int SamePlusTriggers { get; set; }
    public int UltraHardWins { get; set; }
    public int TimesStartedOver { get; set; }
    public int EasyWins { get; set; }
    public int MediumWins { get; set; }
    public int HardWins { get; set; }
    public int CardsStolen { get; set; }
    public int FallenAces { get; set; }
    public int SuddenDeathCount { get; set; }
    public List<int> CollectedCardIds { get; set; } = new List<int> { 1, 2, 3, 4, 5 }; // default starter deck

    // Win % of decisive games (draws excluded from the denominator) — was
    // independently reimplemented identically in MainWindow.axaml.cs (and again on
    // mac/iOS, now fixed to share shared/Honeycomb/Models/HoneycombStats.swift's
    // winRate). Now the one shared source all three read.
    public double WinRate
    {
        get
        {
            int decisiveGames = GamesPlayed - MatchesDrawn;
            return decisiveGames > 0 ? 100.0 * MatchesWon / decisiveGames : 0.0;
        }
    }

    public void RecordGame(bool won, bool drawn, int captures, int sessionCombos, bool flawless, HoneycombDifficulty difficulty, int fallenAceCaptures)
    {
        GamesPlayed++;
        CardsCaptured += captures;
        SamePlusTriggers += sessionCombos;
        FallenAces += fallenAceCaptures;

        if (drawn)
        {
            MatchesDrawn++;
            CurrentWinStreak = 0;
        }
        else if (won)
        {
            MatchesWon++;
            CurrentWinStreak++;
            if (CurrentWinStreak > LongestWinStreak)
            {
                LongestWinStreak = CurrentWinStreak;
            }

            if (flawless) FlawlessVictories++;

            switch (difficulty)
            {
                case HoneycombDifficulty.Easy: EasyWins++; break;
                case HoneycombDifficulty.Medium: MediumWins++; break;
                case HoneycombDifficulty.Hard: HardWins++; break;
                case HoneycombDifficulty.UltraHard: UltraHardWins++; break;
            }
        }
        else
        {
            MatchesLost++;
            CurrentWinStreak = 0;
        }
    }
}

File: android/app/src/main/java/com/leah/honeycomb/honeycomb/HoneycombStats.kt
package com.leah.honeycomb.honeycomb

import kotlinx.serialization.Serializable

// Ported 1:1 from shared/Honeycomb/Models/HoneycombStats.swift.
@Serializable
data class HoneycombStats(
    val gamesPlayed: Int = 0,
    val matchesWon: Int = 0,
    val matchesLost: Int = 0,
    val matchesDrawn: Int = 0,
    val cardsCaptured: Int = 0,

    val currentWinStreak: Int = 0,
    val longestWinStreak: Int = 0,
    val flawlessVictories: Int = 0,
    val samePlusTriggers: Int = 0,
    val ultraHardWins: Int = 0,
    val timesStartedOver: Int = 0,
    val easyWins: Int = 0,
    val mediumWins: Int = 0,
    val hardWins: Int = 0,
    val cardsStolen: Int = 0,
    val fallenAces: Int = 0,
    val suddenDeathCount: Int = 0
) {
    // Win % of decisive games (draws excluded from the denominator) — matches
    // HoneycombStats.swift's winRate, the single shared source Mac/iOS both read.
    val winRate: Double
        get() {
            val decisiveGames = gamesPlayed - matchesDrawn
            return if (decisiveGames > 0) matchesWon.toDouble() / decisiveGames.toDouble() else 0.0
        }

    fun recordGame(
        won: Boolean,
        drawn: Boolean,
        captures: Int,
        sessionCombos: Int,
        flawless: Boolean,
        difficulty: HoneycombDifficulty? = null,
        fallenAceCaptures: Int = 0
    ): HoneycombStats {
        var result = copy(
            gamesPlayed = gamesPlayed + 1,
            cardsCaptured = cardsCaptured + captures,
            samePlusTriggers = samePlusTriggers + sessionCombos,
            fallenAces = fallenAces + fallenAceCaptures
        )

        result = when {
            drawn -> result.copy(matchesDrawn = result.matchesDrawn + 1, currentWinStreak = 0)
            won -> {
                val newStreak = result.currentWinStreak + 1
                var won = result.copy(
                    matchesWon = result.matchesWon + 1,
                    currentWinStreak = newStreak,
                    longestWinStreak = maxOf(result.longestWinStreak, newStreak),
                    flawlessVictories = result.flawlessVictories + if (flawless) 1 else 0
                )
                won = when (difficulty) {
                    HoneycombDifficulty.Easy -> won.copy(easyWins = won.easyWins + 1)
                    HoneycombDifficulty.Medium -> won.copy(mediumWins = won.mediumWins + 1)
                    HoneycombDifficulty.Hard -> won.copy(hardWins = won.hardWins + 1)
                    HoneycombDifficulty.UltraHard -> won.copy(ultraHardWins = won.ultraHardWins + 1)
                    null -> won
                }
                won
            }
            else -> result.copy(matchesLost = result.matchesLost + 1, currentWinStreak = 0)
        }

        return result
    }
}
